tests/php/partner/WHMCS/JetpackModuleTest.php

```diff
@@ -0,0 +1,118 @@
+<?php
+
+if (!defined('WHMCS')) {
+	define('WHMCS', true);
+}
+
+require_once __DIR__ . '/../../../../partner/WHMCS/modules/servers/jetpack/jetpack.php';
+
+
+use PHPUnit\Framework\TestCase as TestCase;
+
+class JetpackModuleTest extends TestCase {
+
+
+	protected $moduleName = 'jetpack';
+
+	protected $allowed_plans = ['free', 'personal', 'premium', 'professional'];
+
+	protected $params = [
+		'customfields' => [
+			'Plan' => 'Personal',
+			'Site URL' => 'testdomain.com',
+			'Local User' => 'admin',
+			'jetpack_provisioning_details' => ''
+		],
+		'configoption1' => 12345,
+		'configoption2' => 'aclientsecret'
+	];
+
+	/**
+	 * Verify that the required core functions are present in the module
+	 */
+	public function testRequiredCoreFunctionsExists()
+	{
+		$this->assertTrue(function_exists($this->moduleName . '_ConfigOptions'));
+		$this->assertTrue(function_exists($this->moduleName . '_CreateAccount'));
+		$this->assertTrue(function_exists($this->moduleName . '_TerminateAccount'));
+	}
+
+	/**
+	 * Test the validate required fields function with good params
+	 */
+	public function test_validate_required_fields()
+	{
+		$this->assertTrue(validate_required_fields(array_merge($this->params)));
+	}
+
+	/**
+	 * Test that an error string is returned if the client id is missing
+	 * in the config options
+	 */
+	public function test_validate_required_fields_missing_client_id()
```

Comment:
Thanks for adding those tests! I like that each specific situation is in its own test method.